packages/discovery-provider/src/queries/comments/utils.py

```diff
@@ -489,22 +525,55 @@ def format_comments(
     Returns:
         List of formatted comments
     """
-    from src.queries.comments.replies import get_replies
-
-    # Pre-fetch all replies if needed
-    replies_map = {}
-    if include_replies:
+    # If we don't need to include replies, just format the comments directly
+    if not include_replies:
+        formatted_comments = []
         for comment_data in comments:
-            comment = comment_data[0]  # First element is always the comment
-            # For tombstone comments (deleted with replies), we need to fetch replies
-            # even if the comment is deleted
-            replies = get_replies(
-                session, comment.comment_id, current_user_id, artist_id, limit=None
+            # Handle different tuple structures from different queries
```

Comment:
wow yeah these queries are complicated....
Wonder if we should eventually make these more WET instead of trying to handle a bunch of different cases together in the same place 😅

Reply:
yeah part of my work recently was to bring them together since there is so much shared logic around deleted, reported, muted etc, but starting to think this may have made things more complicated